samples/iolib/src/main/cpp/player/SampleBuffer.h

```diff
@@ -33,21 +31,22 @@ struct AudioProperties {
 
 class SampleBuffer {
 public:
-    SampleBuffer() : mSampleData(nullptr), mNumSampleFrames(0) {};
+    SampleBuffer() : mNumSampleFrames(0) {};
+    ~SampleBuffer() { unloadSampleData(); }
 
     // Data load/unload
-    void loadSampleData(WavStreamReader* reader);
+    void loadSampleData(parselib::WavStreamReader* reader);
     void unloadSampleData();
 
     virtual AudioProperties getProperties() const { return mAudioProperties; }
 
-    const float* getSampleData() { return mSampleData; }
+    std::shared_ptr<float*> getSampleData() { return mSampleData; }
```

Comment:
float[]

Reply:
+1

Reply:
I haven't been able to get the compiler to accept that.